File: app/src/main/java/ar/gob/coronavirus/flujos/BaseViewModel.java
package ar.gob.coronavirus.flujos;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

public class BaseViewModel extends ViewModel {
    CompositeDisposable disposable = new CompositeDisposable();

    protected MutableLiveData<Boolean> resultadoDialogoDePermisoDeUbicacionLivaData = new MutableLiveData<>();
    protected MutableLiveData<Integer> lanzarDialogoPermisosLocalizacionLiveData = new MutableLiveData<>();

    public BaseViewModel() {
    }

    public MutableLiveData<Boolean> obtenerResultadoDialogoDePermisoDeUbicacionLivaData() {
        return resultadoDialogoDePermisoDeUbicacionLivaData;
    }

    public MutableLiveData<Integer> obtenerLanzarDialogoPermisosLocalizacionLiveData() {
        return lanzarDialogoPermisosLocalizacionLiveData;
    }

    public void setResultadoDialogoCustomPermisoDeUbicacion(boolean puedeMostrarDialogoNativo) {
        resultadoDialogoDePermisoDeUbicacionLivaData.setValue(puedeMostrarDialogoNativo);
    }

    public void lanzarDialogoPermisosLocalizacion(int tipoDePermisoDeUbicacion) {
        lanzarDialogoPermisosLocalizacionLiveData.setValue(tipoDePermisoDeUbicacion);
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        disposable.dispose();
    }

    protected void addDisposable(Disposable disposable) {
        this.disposable.add(disposable);
    }
}

File: app/src/main/java/ar/gob/coronavirus/flujos/BaseViewModel.kt
package ar.gob.coronavirus.flujos

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

open class BaseViewModel : ViewModel() {
    private val disposable = CompositeDisposable()

    private val _locationPermissionResultLiveData = MutableLiveData<Boolean>()
    val locationPermissionResultLiveData: LiveData<Boolean>
        get() = _locationPermissionResultLiveData
    private val _showLocationPermissionDialogLiveData = MutableLiveData<Int>()
    val showLocationPermissionDialogLiveData: LiveData<Int>
        get() = _showLocationPermissionDialogLiveData

    fun setPermissionDialogResult(result: Boolean) {
        _locationPermissionResultLiveData.value = result
    }

    fun showLocationPermission(permissionType: Int) {
        _showLocationPermissionDialogLiveData.value = permissionType
    }

    override fun onCleared() {
        super.onCleared()
        disposable.dispose()
    }

    protected fun addDisposable(disposable: Disposable) {
        this.disposable.add(disposable)
    }
}